### PR TITLE
Transaction pre validation endpoint

### PR DESCRIPTION
**Pull request by PawelPawelec-RDX**
There are many todos to fill in description, i'll fill that once we agree how contract should look like.

### CODE REVIEW

File: src/RadixDlt.NetworkGateway.GatewayApi/gateway-api-schema.yaml [hunk 1]
```diff
@@ -2387,7 +2411,90 @@ components:
...
+    AccountDepositPreValidationDecidingFactors:
+      description: Deciding factors used to calculate response.
+      type: object
+      required:
+        - default_deposit_rule
+      properties:
+        is_badge_authorized_depositor:
+          description: Whether the input badge belongs to the account's set of authorized depositors. This field will only be present if any badge was passed in the request.
+          type: boolean
+          nullable: true
```
Comment (krzlabrdx) [inline src/RadixDlt.NetworkGateway.GatewayApi/gateway-api-schema.yaml:2433]: I think it's enough to not enlist this type in `required`?

File: src/RadixDlt.NetworkGateway.GatewayApi/gateway-api-schema.yaml [hunk 2]
```diff
@@ -2387,7 +2411,90 @@ components:
...
+
+    TransactionAccountDepositPreValidationAuthorizedDepositorBadge:
+      type: object
+      required:
+        - badge_type
+        - resource_address
+      properties:
+        badge_type:
+          $ref: "#/components/schemas/AccountAuthorizedDepositorBadgeType"
+        resource_address:
```
Comment (krzlabrdx) [inline src/RadixDlt.NetworkGateway.GatewayApi/gateway-api-schema.yaml:2477]: I believe we were supposed to move `resource_address` to derived types as the fact that it exists in all derived types today is nothing more but a coincidence.

File: src/RadixDlt.NetworkGateway.GatewayApi/gateway-api-schema.yaml [hunk 3]
```diff
@@ -3011,6 +3118,44 @@ components:
...
+              $ref: "#/components/schemas/Address"
+            resource_addresses:
+              type: array
+              items:
+                $ref: "#/components/schemas/Address"
+            badge:
+              $ref: "#/components/schemas/TransactionAccountDepositPreValidationAuthorizedDepositorBadge"
+          example:
+            account_address: "<account_address>"
+            resource_addresses: ["<resource_address>"]
```
Comment (krzlabrdx) [inline src/RadixDlt.NetworkGateway.GatewayApi/gateway-api-schema.yaml:3138]: IIRC we use hyphens not underscores for those placeholders?

File: src/RadixDlt.NetworkGateway.PostgresIntegration/Services/QueryHelper.cs [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+    internal static async Task<TEntity> GetEntity<TEntity>(ReadOnlyDbContext dbContext, EntityAddress address, GatewayApiSdk.Model.LedgerState ledgerState, CancellationToken token)
+        where TEntity : Entity
+    {
+        var entity = await dbContext
+            .Entities
+            .Where(e => e.FromStateVersion <= ledgerState.StateVersion)
+            .AnnotateMetricName()
+            .FirstOrDefaultAsync(e => e.Address == address, token);
+
+        if (entity == null)
```
Comment (krzlabrdx) [inline src/RadixDlt.NetworkGateway.PostgresIntegration/Services/QueryHelper.cs:98]: Just remember that we can't use it anywhere virtual accounts/identities are expected! Are we OK with that?

**Review by krzlabrdx — CHANGES_REQUESTED** [review]

File: src/RadixDlt.NetworkGateway.GatewayApi/gateway-api-schema.yaml (hunk 1 above)
Comment (PawelPawelec-RDX) [reply src/RadixDlt.NetworkGateway.GatewayApi/gateway-api-schema.yaml:2433]: it defaulted to false without it. I think it handles differently simple types.

File: src/RadixDlt.NetworkGateway.GatewayApi/gateway-api-schema.yaml (hunk 2 above)
Comment (PawelPawelec-RDX) [reply src/RadixDlt.NetworkGateway.GatewayApi/gateway-api-schema.yaml:2477]: yes, but couldn't force openAPI generator to properly generate classes if that was inside child classes.

File: src/RadixDlt.NetworkGateway.PostgresIntegration/Services/QueryHelper.cs (hunk 4 above)
Comment (PawelPawelec-RDX) [reply src/RadixDlt.NetworkGateway.PostgresIntegration/Services/QueryHelper.cs:98]: yes, we don't expect virtual entities in places where I've used it.

**Review by krzlabrdx — APPROVED** [review]

**Pull request merged**